File: Examples/FSharp/WebSocketApi/AccountTrade/AllOcoOrders_Example.fs
open System
open System.Collections.Generic
open System.IO;
open System.Net
open System.Net.Http
open System.Threading;
open System.Threading.Tasks
open Microsoft.Extensions.Logging
open Binance.Common
open Binance.Spot
open Binance.Spot.Models


[<EntryPoint>]
let main argv =
    let loggerFactory = LoggerFactory.Create(fun (builder:ILoggingBuilder) -> 
        builder.AddConsole() |> ignore
    )
    let logger = loggerFactory.CreateLogger()

    let privateKey = File.ReadAllText("/Users/john/ssl/Private_key.pem");

    let websocket = new WebSocketApi("wss://testnet.binance.vision/ws-api/v3", "apiKey", new BinanceRsa(privateKey));

    let onlyOneMessage = new TaskCompletionSource<string>();

    websocket.OnMessageReceived(
        (fun data ->
            onlyOneMessage.SetResult(data)
            Task.CompletedTask)
    );

    websocket.ConnectAsync() |> Async.AwaitTask |> Async.RunSynchronously

    websocket.AccountTrade.AllOcoOrdersAsync(startTime = 0, recvWindow = 6000, requestId = 123) |> Async.AwaitTask |> Async.RunSynchronously

    let message = onlyOneMessage.Task |> Async.AwaitTask |> Async.RunSynchronously

    logger.LogInformation(message);

    websocket.DisconnectAsync() |> Async.AwaitTask |> Async.RunSynchronously
    
    0
    

File: Examples/CSharp/WebSocketApi/AccountTrade/AllOcoOrders_Example.cs
namespace Binance.Spot.WebSocketApiExamples
{
    using System.Threading;
    using System.Threading.Tasks;
    using Binance.Common;
    using Binance.Spot;
    using Microsoft.Extensions.Logging;

    public class AllOcoOrders_Example
    {
        public static async Task Main(string[] args)
        {
            using var loggerFactory = LoggerFactory.Create(builder =>
            {
                builder.AddConsole();
            });
            ILogger logger = loggerFactory.CreateLogger<AllOcoOrders_Example>();

            string apiSecret = "apiSecret";
            var websocket = new WebSocketApi("wss://testnet.binance.vision/ws-api/v3", "apiKey", new BinanceHmac(apiSecret));

            websocket.OnMessageReceived(
                async (data) =>
            {
                logger.LogInformation(data);
                await Task.CompletedTask;
            }, CancellationToken.None);

            await websocket.ConnectAsync(CancellationToken.None);

            await websocket.AccountTrade.AllOcoOrdersAsync(1);
            await websocket.AccountTrade.AllOcoOrdersAsync(startTime: 0, recvWindow: 6000, requestId: 123);

            await Task.Delay(3000);
            logger.LogInformation("Disconnect with WebSocket Server");
            await websocket.DisconnectAsync(CancellationToken.None);
        }
    }
}